zhaquirks/tuya/ts0601_valve.py

```diff
@@ -493,3 +502,97 @@ class GiexValveVar02(CustomDevice):
             }
         }
     }
+
+
+class GiexIrrigationStatus(t.enum8):
+    """Giex Irrigation Status Enum."""
+
+    Manual = 0x00
+    Auto = 0x01
+    Idle = 0x02
+
+
+(
+    TuyaQuirkBuilder("_TZE284_8zizsafo", "TS0601")  # Giex GX04
+    .tuya_battery(dp_id=59, power_cfg=TuyaPowerConfigurationCluster4AA)
+    .tuya_switch(
+        dp_id=1,
+        attribute_name="valve_one_on_off",
+        entity_type=EntityType.STANDARD,
+        translation_key="valve_one_on_off",
+        fallback_name="Valve 1",
+    )
+    .tuya_switch(
+        dp_id=2,
+        attribute_name="valve_two_on_off",
+        entity_type=EntityType.STANDARD,
+        translation_key="valve_two_on_off",
+        fallback_name="Valve 2",
+    )
+    .tuya_number(
+        dp_id=13,
+        attribute_name="valve_one_countdown",
+        type=t.uint16_t,
+        device_class=SensorDeviceClass.DURATION,
+        unit=UnitOfTime.MINUTES,
+        min_value=0,
+        max_value=1440,
+        step=1,
+        translation_key="valve_one_countdown",
+        fallback_name="Irrigation time 1",
+    )
+    .tuya_number(
+        dp_id=14,
+        attribute_name="valve_two_countdown",
+        type=t.uint16_t,
+        device_class=SensorDeviceClass.DURATION,
+        unit=UnitOfTime.MINUTES,
+        min_value=0,
+        max_value=1440,
+        step=1,
+        translation_key="valve_two_countdown",
+        fallback_name="Irrigation time 2",
+    )
+    .tuya_sensor(
+        dp_id=25,
+        attribute_name="valve_one_duration",
+        type=t.uint32_t,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        unit=UnitOfTime.SECONDS,
+        entity_type=EntityType.STANDARD,
+        translation_key="irrigation_duration",
```

Comment:
This `translation_key` is the same here and below.
I'm actually not sure what currently happens if `fallback_name` and/or `translation_key` is the same for multiple entities and what the proper way to handle this is. I'd guess HA would want the same translation key for all entities that are the same..?
Ideally, HA would add `2` at the end. Have you tried this with either a duplicate fallback name (or a duplicate translation key with the translations added manually to HA)?